datafusion/physical-plan/src/aggregates/group_values/group_column.rs

```diff
@@ -58,6 +63,17 @@ pub trait GroupColumn: Send + Sync {
     fn equal_to(&self, lhs_row: usize, array: &ArrayRef, rhs_row: usize) -> bool;
     /// Appends the row at `row` in `array` to this builder
     fn append_val(&mut self, array: &ArrayRef, row: usize);
+
+    fn vectorized_equal_to(
```

Comment:
Are these vectorized_xxx SIMD? or they are just processing the rows all at once?

Reply:
It only switch the process logic from `row oriented` to `col oriented`.

The original logic is like:
```
eq col1[0], col2[0], col3[0]
eq col1[1], col2[1], col3[1]
...
eq col1[n], col2[n], col3[n]
```

And current logic:
```
eq col1[0] ~ col1[n]
eq col2[0] ~ col2[n]
eq col3[0] ~ col3[[n]
```

Reply:
It can get performance improvement due to eliminating the unnecessary downcasting for each row.
And maybe can reduce the random memory accesses.

Reply:
It is really a good point to improve later.
After refactoring to `col oriented` approach, more things we can try.